File: MiniDOMBase/DMD/Sistema/Calendar/Activities/CCalendarActivityCursor.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica

Partial Class Sistema


   

    ''' <summary>
    ''' Cursore sulle attività del calendario
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CCalendarActivityCursor
        Inherits DBObjectCursor(Of CCalendarActivity)

        Private m_Categoria As New CCursorFieldObj(Of String)("Categoria")
        Private m_Descrizione As New CCursorFieldObj(Of String)("Descrizione")
        Private m_DataInizio As New CCursorField(Of Date)("DataInizio")
        Private m_DataFine As New CCursorField(Of Date)("DataFine")
        Private m_StatoAttivita As New CCursorField(Of StatoAttivita)("StatoAttivita")
        Private m_IDOperatore As New CCursorField(Of Integer)("Operatore")
        Private m_NomeOperatore As New CCursorFieldObj(Of String)("NomeOperatore")
        Private m_IDAssegnatoA As New CCursorField(Of Integer)("IDAssegnatoA")
        Private m_NomeAssegnatoA As New CCursorFieldObj(Of String)("NomeAssegnatoA")
        Private m_Note As New CCursorFieldObj(Of String)("Note")
        Private m_Luogo As New CCursorFieldObj(Of String)("Luogo")
        Private m_Promemoria As New CCursorField(Of Integer)("Promemoria")
        Private m_Ripetizione As New CCursorField(Of Integer)("Ripetizione")
        Private m_GiornataIntera As New CCursorField(Of Boolean)("GiornataIntera")
        Private m_Flags As New CCursorField(Of Integer)("Flags")
        Private m_IDPersona As New CCursorField(Of Integer)("IDPersona")
        Private m_NomePersona As New CCursorFieldObj(Of String)("NomePersona")
        Private m_ProviderName As New CCursorFieldObj(Of String)("ProviderName")
        Private m_Priorita As New CCursorField(Of Integer)("Priorita")

        Public Sub New()
        End Sub

        Public ReadOnly Property Priorita As CCursorField(Of Integer)
            Get
                Return Me.m_Priorita
            End Get
        End Property

        Public ReadOnly Property Categoria As CCursorFieldObj(Of String)
            Get
                Return Me.m_Categoria
            End Get
        End Property

        Public ReadOnly Property Descrizione As CCursorFieldObj(Of String)
            Get
                Return Me.m_Descrizione
            End Get
        End Property

        Public ReadOnly Property DataInizio As CCursorField(Of Date)
            Get
                Return Me.m_DataInizio
            End Get
        End Property

        Public ReadOnly Property DataFine As CCursorField(Of Date)
            Get
                Return Me.m_DataFine
            End Get
        End Property

        Public ReadOnly Property StatoAttivita As CCursorField(Of StatoAttivita)
            Get
                Return Me.m_StatoAttivita
            End Get
        End Property

        Public ReadOnly Property IDOperatore As CCursorField(Of Integer)
            Get
                Return Me.m_IDOperatore
            End Get
        End Property

        Public ReadOnly Property NomeOperatore As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeOperatore
            End Get
        End Property


        Public ReadOnly Property IDAssegnatoA As CCursorField(Of Integer)
            Get
                Return Me.m_IDAssegnatoA
            End Get
        End Property

        Public ReadOnly Property NomeAssegnatoA As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeAssegnatoA
            End Get
        End Property

        Public ReadOnly Property Note As CCursorFieldObj(Of String)
            Get
                Return Me.m_Note
            End Get
        End Property

        Public ReadOnly Property Luogo As CCursorFieldObj(Of String)
            Get
                Return Me.m_Luogo
            End Get
        End Property

        Public ReadOnly Property Promemoria As CCursorField(Of Integer)
            Get
                Return Me.m_Promemoria
            End Get
        End Property

        Public ReadOnly Property Ripetizione As CCursorField(Of Integer)
            Get
                Return Me.m_Ripetizione
            End Get
        End Property

        Public ReadOnly Property GiornataIntera As CCursorField(Of Boolean)
            Get
                Return Me.m_GiornataIntera
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of Integer)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public ReadOnly Property IDPersona As CCursorField(Of Integer)
            Get
                Return Me.m_IDPersona
            End Get
        End Property

        Public ReadOnly Property NomePersona As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomePersona
            End Get
        End Property

        Public ReadOnly Property ProviderName As CCursorFieldObj(Of String)
            Get
                Return Me.m_ProviderName
            End Get
        End Property

        Protected Overrides Function GetModule() As CModule
            Return DateUtils.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_CalendarActivities"
        End Function

        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return Databases.APPConn
        End Function


    End Class
 
End Class

File: MiniDOMBase/DMD/Sistema/Calendar/Activities/CCalendarActivityCursor.cs
using System;
using System.Collections;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using static minidom.Sistema;
using static minidom.Anagrafica;
using minidom.repositories;
using System.Collections.Generic;


namespace minidom
{
    public partial class Sistema
    {


        /// <summary>
        /// Cursore sulle attività del calendario
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CCalendarActivityCursor
            : minidom.Databases.DBObjectCursor<CCalendarActivity>
        {
            private DBCursorStringField m_Categoria = new DBCursorStringField("Categoria");
            private DBCursorStringField m_Descrizione = new DBCursorStringField("Descrizione");
            private DBCursorField<DateTime> m_DataInizio = new DBCursorField<DateTime>("DataInizio");
            private DBCursorField<DateTime> m_DataFine = new DBCursorField<DateTime>("DataFine");
            private DBCursorField<StatoAttivita> m_StatoAttivita = new DBCursorField<StatoAttivita>("StatoAttivita");
            private DBCursorField<int> m_IDOperatore = new DBCursorField<int>("Operatore");
            private DBCursorStringField m_NomeOperatore = new DBCursorStringField("NomeOperatore");
            private DBCursorField<int> m_IDAssegnatoA = new DBCursorField<int>("IDAssegnatoA");
            private DBCursorStringField m_NomeAssegnatoA = new DBCursorStringField("NomeAssegnatoA");
            private DBCursorStringField m_Note = new DBCursorStringField("Note");
            private DBCursorStringField m_Luogo = new DBCursorStringField("Luogo");
            private DBCursorField<int> m_Promemoria = new DBCursorField<int>("Promemoria");
            private DBCursorField<int> m_Ripetizione = new DBCursorField<int>("Ripetizione");
            private DBCursorField<bool> m_GiornataIntera = new DBCursorField<bool>("GiornataIntera");
            private DBCursorField<CalendarActivityFlags> m_Flags = new DBCursorField<CalendarActivityFlags>("Flags");
            private DBCursorField<int> m_IDPersona = new DBCursorField<int>("IDPersona");
            private DBCursorStringField m_NomePersona = new DBCursorStringField("NomePersona");
            private DBCursorStringField m_ProviderName = new DBCursorStringField("ProviderName");
            private DBCursorField<int> m_Priorita = new DBCursorField<int>("Priorita");

            /// <summary>
            /// Costruttore
            /// </summary>
            public CCalendarActivityCursor()
            {
            }

            /// <summary>
            /// Priorita
            /// </summary>
            public DBCursorField<int> Priorita
            {
                get
                {
                    return m_Priorita;
                }
            }

            /// <summary>
            /// Categoria
            /// </summary>
            public DBCursorStringField Categoria
            {
                get
                {
                    return m_Categoria;
                }
            }

            /// <summary>
            /// Descrizione
            /// </summary>
            public DBCursorStringField Descrizione
            {
                get
                {
                    return m_Descrizione;
                }
            }

            /// <summary>
            /// DataInizio
            /// </summary>
            public DBCursorField<DateTime> DataInizio
            {
                get
                {
                    return m_DataInizio;
                }
            }

            /// <summary>
            /// DataFine
            /// </summary>
            public DBCursorField<DateTime> DataFine
            {
                get
                {
                    return m_DataFine;
                }
            }

            /// <summary>
            /// StatoAttivita
            /// </summary>
            public DBCursorField<StatoAttivita> StatoAttivita
            {
                get
                {
                    return m_StatoAttivita;
                }
            }

            /// <summary>
            /// IDOperatore
            /// </summary>
            public DBCursorField<int> IDOperatore
            {
                get
                {
                    return m_IDOperatore;
                }
            }

            /// <summary>
            /// NomeOperatore
            /// </summary>
            public DBCursorStringField NomeOperatore
            {
                get
                {
                    return m_NomeOperatore;
                }
            }

            /// <summary>
            /// IDAssegnatoA
            /// </summary>
            public DBCursorField<int> IDAssegnatoA
            {
                get
                {
                    return m_IDAssegnatoA;
                }
            }

            /// <summary>
            /// NomeAssegnatoA
            /// </summary>
            public DBCursorStringField NomeAssegnatoA
            {
                get
                {
                    return m_NomeAssegnatoA;
                }
            }

            /// <summary>
            /// Note
            /// </summary>
            public DBCursorStringField Note
            {
                get
                {
                    return m_Note;
                }
            }

            /// <summary>
            /// Luogo
            /// </summary>
            public DBCursorStringField Luogo
            {
                get
                {
                    return m_Luogo;
                }
            }

            /// <summary>
            /// Promemoria
            /// </summary>
            public DBCursorField<int> Promemoria
            {
                get
                {
                    return m_Promemoria;
                }
            }

            /// <summary>
            /// Ripetizione
            /// </summary>
            public DBCursorField<int> Ripetizione
            {
                get
                {
                    return m_Ripetizione;
                }
            }

            /// <summary>
            /// GiornataIntera
            /// </summary>
            public DBCursorField<bool> GiornataIntera
            {
                get
                {
                    return m_GiornataIntera;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<CalendarActivityFlags> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            /// <summary>
            /// IDPersona
            /// </summary>
            public DBCursorField<int> IDPersona
            {
                get
                {
                    return m_IDPersona;
                }
            }

            /// <summary>
            /// NomePersona
            /// </summary>
            public DBCursorStringField NomePersona
            {
                get
                {
                    return m_NomePersona;
                }
            }

            /// <summary>
            /// ProviderName
            /// </summary>
            public DBCursorStringField ProviderName
            {
                get
                {
                    return m_ProviderName;
                }
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Sistema.Calendar; //.Module;
            }

            //public override string GetTableName()
            //{
            //    return "tbl_CalendarActivities";
            //}

            //protected internal override CDBConnection GetConnection()
            //{
            //    return APPConn;
            //}
        }
    }
}